llvm/docs/DirectX/DXContainer.rst

```diff
@@ -400,3 +400,273 @@ SFI0 Part
+   table, constants, CBV, SRV, UAV).
+#. **ShaderVisibility**: Specifies which shader stages can access this parameter (e.g., all stages, 
+   vertex shader only, pixel shader only).
+#. **ParameterOffset**: Byte offset to the specific parameter data structure 
+   for this entry.
+
+The header uses a parameter type field rather than encoding the version of the parameter through 
+size, allowing for a more explicit representation of the parameter's nature.
+
+Root Parameter Types
+''''''''''''''''''''
+This section describes the representation of each root parameter type.
+
+Root Constants
+^^^^^^^^^^^^^^
+
+.. code-block:: c
+
+  struct RootConstants {
+    uint32_t ShaderRegister;
+    uint32_t RegisterSpace;
+    uint32_t Num32BitValues;
+  };
+
+The ``RootConstants`` structure represents inline root constants that are directly embedded in the root 
+signature and passed to the shader without requiring a constant buffer resource:
+
+#. **ShaderRegister**: The shader register (b#) where these constants are bound.
+#. **RegisterSpace**: The register space used for the binding.
+#. **Num32BitValues**: The number of 32-bit values included in this constant buffer.
+
+Root constants provide a fast way to pass small amounts of data directly to the shader without the 
+overhead of creating and binding a constant buffer resource.
+
+Root Descriptor
+^^^^^^^^^^^^^^^
+
+Root descriptors provide a mechanism for binding individual resources to shader stages in the Direct3D 12 
+rendering pipeline. They allow applications to specify how shader stages access specific GPU resources.
+
+.. code-block:: c
+
+   enum RootDescriptorFlags {
+      None = 0,
+      DataVolatile = 0x2,
+      DataStaticWhileSetAtExecute = 0x4,
+      DataStatic = 0x8,
+   }
+
+   // Version 1.0 Root Descriptor
+   struct RootDescriptor_V1_0 {
+      uint32_t ShaderRegister;
+      uint32_t RegisterSpace;
+   };
+   
+   // Version 1.1 Root Descriptor
+   struct RootDescriptor_V1_1 {
+      uint32_t ShaderRegister;
+      uint32_t RegisterSpace;      
+      // Bitfield of flags from the Flags enum
+      uint32_t Flags;
+   };
+
+Version 1.1 of Root Descriptors has introduced some flags that can hint the drivers into
+performing further code optimizations. For details, check
+`Direct X documentation <https://learn.microsoft.com/en-us/windows/win32/direct3d12/root-signature-version-1-1#static-and-volatile-flags>`_.
+
+Version 1.0 Root Descriptor
+"""""""""""""""""""""""""""
+The Version 1.0 RootDescriptor_V1_0 provides basic resource binding:
+
+#. **ShaderRegister**: The shader register where the descriptor is bound.
+#. **RegisterSpace**: The register space used for the binding.
+
+Version 1.1 Root Descriptor
+"""""""""""""""""""""""""""
+The Version 1.1 RootDescriptor_V1_1 extends the base structure with the following additional fields:
+
+#. **Flags**: Provides additional metadata about the descriptor's usage pattern.
```

Comment:
Given that version 1.1 only seems to have added a flags field, I think it might be clearer to describe the 1.1 format and then have a note about the flags that it wasn't present in 1.0